### DISCUSSION TITLE
compile-option USE_SML_M breaks compile with obsolete library "ams"

### DISCUSSION THREAD

**Discussion by JoeRu** [arendst/Tasmota#19744] [Q&A] [answered]
### PROBLEM DESCRIPTION
Download latest v13.1. tag from github.
Followed description of documentation here: 
https://tasmota.github.io/docs/Smart-Meter-Interface/
Use Visual Code and Platformio to compile tasmota.

Compile fails with 
```
lib\lib_div\ams\GcmParser.cpp:7:23: fatal error: t_bearssl.h: No such file or directory

*******************************************************************
* Looking for t_bearssl.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:t_bearssl.h"
* Web  > https://registry.platformio.org/search?q=header:t_bearssl.h
*
*******************************************************************

 #include <t_bearssl.h>
                       ^
compilation terminated.
*** [.pio\build\tasmota-sensors\libcb3\ams\GcmParser.cpp.o] Error 1
```

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [ X] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [X ] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [X ] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [X ] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ ] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4) 
No Discord account
- [ ] Device used (e.g., Sonoff Basic): _____
- [ ] Tasmota binary firmware version number used: tag v13.1.0
  - [ ] Pre-compiled
  - [X] Self-compiled
- [X] Flashing tools used: VisualCode with PlatformIO
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:


```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:


```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior:_
Download git Version - switch to v13.1.0. 
 Add the following to user_config_override.h:

```
#ifndef USE_SML_M
#define USE_SML_M
#endif
```
as described here https://tasmota.github.io/docs/Smart-Meter-Interface/

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

A error free compilation.

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._
I suppose the used ams library or the used bearssl library is not maintained any longer.

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by barbudor** [comment, marked as answer]
> *** [.pio\build\tasmota-sensors\libcb3\ams\GcmParser.cpp.o] Error 1

Behave as documented here : https://tasmota.github.io/docs/Compile-your-build/#customize-your-build

![image](https://github.com/arendst/Tasmota/assets/2996042/b43da4ff-dab8-495d-a4ee-89ec12f151ca)

**Comment by JoeRu** [comment]
My failure - sorry - using tasmota as described builds regular and without error...
